[WIP] Prototype / hack of integration of DTS into sanitycheck

galak opened the pull request:
This is a wip in progress, first cut/example of using DTS in sanitycheck.  Looking for feedback if this direction seems desirable.  etc.

galak committed:
sanitycheck: Add functions to query device tree for filters

Add the following functions to allow filtering based on device tree

dt_compat_enabled(compat) - Returns true if a device tree node compatible
matches 'compat' and the node is enabled.

dt_alias_exist(alias) - Returns true if a device tree node exists with
'alias' and the node is enabled.

dt_compat_and_alias_enabled - Returns true if a device tree node
compatible matches 'compat' and the node has 'alias' and the node is
enabled.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ulfalizer reviewed (commented):
Haven't looked much at these scripts, but getting rid of `generated_dts_board.conf` seems nice.

pabigot commented on scripts/sanity_chk/expr_parser.py:
```diff
@@ -10,6 +10,11 @@
 import re
 
+ZEPHYR_BASE = os.environ.get("ZEPHYR_BASE")
+sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/dts"))
```
IIRC for portability this should be `os.path.join(ZEPHYR_BASE, "scripts", "dts")`  but there's probably already a lot of code that won't work for `os.path.sep` not equal to `'/'`.  Certainly there's material in `sanitycheck` that should be using `os.path.split` if that's a concern.

pabigot reviewed (approved):
Seems like a clean solution, especially if this means we can put application-specific board conf/overlay files in the test and they get picked up automatically (no mods to `testcase.yaml` or `sample.yaml`).

dbkinder commented on doc/guides/test/sanitycheck.rst:
```diff
@@ -548,15 +439,81 @@ Beside being able to run tests in QEMU and other simulated environments,
+===================================
+
+To use this feature on a single connected device, run sanitycheck with
+the following new options::
```
they won't be "new" for long, so just say "the following options"

dbkinder commented on doc/guides/test/sanitycheck.rst:
```diff
@@ -548,15 +439,81 @@ Beside being able to run tests in QEMU and other simulated environments,
+Executing tests on multiple devices
+===================================
+
+To build and execute tests on multiple devices connected to the host PC a
```
Better would be:
```
To build and execute tests on multiple devices connected to the host PC,
you'll need to create a hardware map with all connected devices and their
details, such as the serial device and their IDs, if available. Create a
hardware map by running this command::
```

dbkinder commented on doc/guides/test/sanitycheck.rst:
```diff
@@ -548,15 +439,81 @@ Beside being able to run tests in QEMU and other simulated environments,
+    serial: /dev/cu.usbmodem0006837593581
+
+
+Any options marked as 'unknown' need to be changed and set with the correct
```
Better would be:
```
You must manually change any options marked as ``unknown``.
In the above example, replace the ``platform`` and ``runner``
names with values appropriate for the connected
hardware. In this example we are using a ``reel_board`` and an
``nrf52840_pca10056`` board, and ``pyocd`` and ``nrfjprog``
runners, respectively::
```

dbkinder commented on doc/guides/test/sanitycheck.rst:
```diff
@@ -548,15 +439,81 @@ Beside being able to run tests in QEMU and other simulated environments,
+
+.. note::
+
+  Currently only boards with support for both pyocd and nrfjprog are supported
```
Do the boards need to support **both pyocd and nrfjprog** ? I'm guessing it's either, and if so, change this to:
```
Currently only boards with support for either pyocd or nrfjprog are supported ...
```

galak committed:
sample/tests: replace DT_ define filters with dt_ functions

convert sample and test yaml filters that utilize a DT_ define to
instead use a dt_ function.  The intent is to remove the Kconfig
generated DT defines and just make directy queries into the device tree.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
sanitycheck: remove import of dt kconfig symbols

All users of dt kconfig symbols for tests and samples are using dt
functions instead so we can remove the support for parsing the
generated file.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
Will open a new PR with the code now that its cleaned up.
